cmd/kubeadm/app/phases/certs/certlist_test.go

```diff
@@ -222,3 +223,110 @@ func parseCertAndKey(basePath string, t *testing.T) (*x509.Certificate, crypto.P
 
 	return parsedCert, certPair.PrivateKey
 }
+
+func Test_createKeyAndCSR(t *testing.T) {
+	dir, err := os.MkdirTemp("", t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	type args struct {
+		kubeadmConfig *kubeadmapi.InitConfiguration
+		cert          *KubeadmCert
+	}
+	tests := []struct {
+		name       string
+		args       args
+		wantErr    bool
+		createfile bool
+	}{
+		{
+			name: "kubeadmConfig is nil",
+			args: args{
+				kubeadmConfig: nil,
+				cert: &KubeadmCert{
+					Name:     "ca",
+					LongName: "self-signed Kubernetes CA to provision identities for other Kubernetes components",
+					BaseName: kubeadmconstants.CACertAndKeyBaseName,
+					config: pkiutil.CertConfig{
+						Config: certutil.Config{
+							CommonName: "kubernetes",
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "cert is nil",
+			args: args{
+				kubeadmConfig: &kubeadmapi.InitConfiguration{
+					NodeRegistration: kubeadmapi.NodeRegistrationOptions{
+						Name: "test-node",
+					},
+					ClusterConfiguration: kubeadmapi.ClusterConfiguration{
+						CertificatesDir: dir,
+					},
+				},
+				cert: nil,
+			},
+			wantErr: true,
+		},
+		{
+			name: "key and CSR do not exist",
+			args: args{
+				kubeadmConfig: &kubeadmapi.InitConfiguration{
+					NodeRegistration: kubeadmapi.NodeRegistrationOptions{
+						Name: "test-node",
+					},
+					ClusterConfiguration: kubeadmapi.ClusterConfiguration{
+						CertificatesDir: dir,
+					},
+				},
+				cert: &KubeadmCert{
```

Comment:
There are many duplicate definitions of `kubeadmConfig` and `KubeadmCert`.
We can define them as variables (`validKubeadmConfig` and `validKubeadmCert` ?) and reference them in the test cases.
WDYT? @my-git9